sumologic/resource_sumologic_cse_log_mapping.go

```diff
@@ -0,0 +1,493 @@
+	unstructuredInputField := CSELogMappingUnstructuredFields{}
+	if len(unstructuredInputFieldSlice) > 0 {
+		unstructuredInputFieldObj := unstructuredInputFieldSlice[0].(map[string]interface{})
+		unstructuredInputField.PatternNames = resourceStringArrayToStringArray(unstructuredInputFieldObj["pattern_names"].([]interface{}))
+	}
+	return unstructuredInputField
+}
+
+func resourceToCSELogMappingLookUp(data interface{}) CSELogMappingLookUp {
+	lookUpSlice := data.([]interface{})
+	lookup := CSELogMappingLookUp{}
+	if len(lookUpSlice) > 0 {
+		lookUpObj := lookUpSlice[0].(map[string]interface{})
+		lookup.Key = lookUpObj["key"].(string)
+		lookup.Value = lookUpObj["value"].(string)
+	}
+	return lookup
+}
+
+func resourceToCSELogMappingField(data interface{}) CSELogMappingField {
+	fieldsSlice := data.([]interface{})
+	field := CSELogMappingField{}
+	if len(fieldsSlice) > 0 {
+		fieldObj := fieldsSlice[0].(map[string]interface{})
+		field.Name = fieldObj["name"].(string)
+		field.Value = fieldObj["value"].(string)
+		field.ValueType = fieldObj["value_type"].(string)
+		field.SkippedValues = resourceStringArrayToStringArray(fieldObj["skipped_values"].([]interface{}))
+		field.DefaultValue = fieldObj["default_value"].(string)
+		field.Format = fieldObj["format"].(string)
+		field.CaseInsensitive = fieldObj["case_insensitive"].(bool)
+		field.AlternateValues = resourceStringArrayToStringArray(fieldObj["alternate_values"].([]interface{}))
+		field.TimeZone = fieldObj["time_zone"].(string)
+		field.SplitDelimiter = fieldObj["split_delimiter"].(string)
+		field.SplitIndex = fieldObj["split_index"].(string)
+		field.FieldJoin = resourceStringArrayToStringArray(fieldObj["field_join"].([]interface{}))
+		field.JoinDelimiter = fieldObj["join_delimiter"].(string)
+		field.FormatParameters = resourceStringArrayToStringArray(fieldObj["format_parameters"].([]interface{}))
+
+		lookUpData := fieldObj["lookup"].([]interface{})
+		var lookup []CSELogMappingLookUp
+		for _, data := range lookUpData {
+			lookup = append(lookup, resourceToCSELogMappingLookUp([]interface{}{data}))
+		}
+
+		field.LookUp = lookup
+
+	}
+	return field
+}
+
+func setLogMapping(d *schema.ResourceData, CSELogMapping *CSELogMapping) {
+	d.Set("name", CSELogMapping.Name)
+	if CSELogMapping.ParentId != "" {
+		d.Set("parent_id", CSELogMapping.ParentId)
+	}
+	d.Set("product_guid", CSELogMapping.ProductGuid)
+	d.Set("record_type", CSELogMapping.RecordType)
+	enabled := *(CSELogMapping.Enabled)
+	d.Set("enabled", enabled)
+	d.Set("relates_entities", CSELogMapping.RelatesEntities)
+	d.Set("skipped_values", CSELogMapping.SkippedValues)
+	setFields(d, CSELogMapping.Fields)
+	setStructuredInputs(d, CSELogMapping.StructuredInputs)
+	if CSELogMapping.UnstructuredFields != nil {
+		setUnStructuredFields(d, *(CSELogMapping.UnstructuredFields))
+	}
+}
+
+func setUnStructuredFields(d *schema.ResourceData, unstructuredFields CSELogMappingUnstructuredFields) {
```

Comment:
Minor nit: The "S" in "Unstructured" should be lowercase.

Reply:
done under:
https://github.com/SumoLogic/terraform-provider-sumologic/pull/284/commits/a994b2d8d74888aad0fa04698f1dd7d8084a3ce6